nixos/modules/services/logging/syslog-ng.nix

```diff
@@ -88,7 +88,7 @@ in {
         StandardOutput = "null";
         Restart = "on-failure";
         ExecStart = "${cfg.package}/sbin/syslog-ng ${concatStringsSep " " syslogngOptions}";
-        ExecReload = "${pkgs.coreutils}/bin/kill -HUP ${pidFile}";
+        ExecReload = "${pkgs.coreutils}/bin/kill -HUP `${pkgs.coreutils}/bin/cat ${pidFile}`";
```

Comment:
Use
```nix
{
  PIDFile = pidFile;
  ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
}
```
instead

(And make sure to test)

Reply:
Both options worked for me.